file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,496 @@
+      "! <p class="shorttext">SAP or RAW Structure</p>
+      "! Is sap or raw structure?
+      sap_raw_structure  TYPE ty_sap_raw_structure_type,
+      " /AIF/FLD_IS_SELECT_OPT
+      "! <p class="shorttext">Select-Option</p>
+      "! Is select-option?
+      is_select_option   TYPE abap_bool,
+      " /AIF/IS_COL
+      "! <p class="shorttext">Display as Column</p>
+      "! Display as column
+      "! $default 'X'
+      is_column_display  TYPE abap_bool,
+    END OF ty_key_field_settings.
+
+  TYPES:
+    "! <p class="shorttext">Key Field Determination by Qualifier</p>
+    "! Key field determination by qualifier
+    BEGIN OF ty_key_field_qualifier,
+      " /AIF/ENABLE_QUALIFIER
+      "! <p class="shorttext">Enable Key Field Qualifier</p>
+      "! Enable key field qualifier?
+      uses_qualifier       TYPE abap_bool,
+      " (type /aif/qualifier_fld_name_kfld)
+      "! <p class="shorttext">Qualifier Field Name</p>
+      "! Qualifier field name
+      qualifier_field_name TYPE c LENGTH 120,
+      " (type /aif/qualifier_operator)
+      "! <p class="shorttext">Qualifier Operator</p>
+      "! Qualifier operator
+      qualifier_operator   TYPE zif_aff_types_v1=>ty_option,
+      " (type /aif/qualifier_value_kfld)
+      "! <p class="shorttext">Qualifier Value</p>
+      "! Qualifier value
+      qualifier_value      TYPE c LENGTH 45,
+    END OF ty_key_field_qualifier.
+
+  TYPES:
+    "! <p class="shorttext">Key Field Selection</p>
+    "! Key field selection type and settings
+    BEGIN OF ty_key_field_selection,
+      " (type /aif/multi_sel_type)
+      "! <p class="shorttext">Selection Type</p>
+      "! Selection type
+      selection_type              TYPE ty_selection_type,
+      " (type /aif/msg_tbl)
+      "! <p class="shorttext">Message Index Table</p>
+      "! Message index table
+      message_index_table         TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/field_number_parent)
+      "! <p class="shorttext">Parent Field Sequence Number</p>
+      "! Parent field sequence number
+      parent_field_sequence       TYPE n LENGTH 3,
+      " (type /aif/icon_l4)
+      "! <p class="shorttext">Icon</p>
+      "! Icon
+      icon                        TYPE c LENGTH 4,
+      " (type /aif/tooltip)
+      "! <p class="shorttext">Icon Tooltip</p>
+      "! Icon tooltip
+      icon_tooltip                TYPE c LENGTH 40,
+      " (type /aif/alertfieldname)
+      "! <p class="shorttext">Alert Recipient Field Name</p>
+      "! Field name in alert recipient assignment table
+      alert_field_name            TYPE c LENGTH 45,
+      " (type /aif/alrt_catfn)
+      "! <p class="shorttext">Category Field Name</p>
+      "! Category field name
+      category_field_name         TYPE c LENGTH 32,
+      " /AIF/IS_HIDDEN, swap to positive stature
+      "! <p class="shorttext">Show Tree Node</p>
+      "! Show tree node in view 1 tree
+      "! $default 'X'
+      is_show_tree_node           TYPE abap_bool,
```

Comment:
We have two verbs directly next to each other. Maybe `show_tree_node` or `is_tree_node_visible`?

Reply:
Hm. I think show_tree_node contradicts your recommendation for boolean, right? I changed it to is_tree_node_visible .